code/modules/projectiles/gun_upgrades.dm

```diff
@@ -0,0 +1,19 @@
+/obj/item/device/recoil_compensator
+	name = "anti-grav stabilizer"
+	desc = "A miniaturized gravity generator, capable of reducing the weight of a gun as well as reducing recoil."
+	icon = 'icons/obj/stock_parts.dmi'
+	icon_state = "quadratic_capacitor"
+
+/obj/item/device/recoil_compensator/afterattack(obj/item/weapon/gun/G, mob/user)
```

Comment:
Needs a check for `proximity` or perhaps for `G in user.contents`. Also I think `!user.incapacitated()` (not sure if click code handles that already but probably not for afterattack).